sql/core/src/main/scala/org/apache/spark/sql/json/JSONRelation.scala

```diff
@@ -105,6 +107,15 @@ private[sql] class JSONRelation(
     jsonSchema
   }
 
+  override private[sql] def buildScan(
+      requiredColumns: Array[String],
+      filters: Array[Filter],
+      inputPaths: Array[String],
+      broadcastedConf: Broadcast[SerializableConfiguration]): RDD[Row] = {
+    refresh()
```

Comment:
`inputPaths` contains the "base" input paths, while the `inputPaths` of the other `buildScan` method in `JSONRelation` contains `FileStatus`es of leaf files under base input paths, which is retrieved from the `FileStatusCache` in `HadoopFsRelation`.